go/chat/identify_notifier.go

```diff
@@ -0,0 +1,43 @@
+package chat
+
+import (
+	"context"
+	"sync"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type IdentifyNotifier struct {
+	libkb.Contextified
+	sync.RWMutex
+	identCache map[string]keybase1.CanonicalTLFNameAndIDWithBreaks
+}
+
+func NewIdentifyNotifier(g *libkb.GlobalContext) *IdentifyNotifier {
+	return &IdentifyNotifier{
+		Contextified: libkb.NewContextified(g),
+		identCache:   make(map[string]keybase1.CanonicalTLFNameAndIDWithBreaks),
+	}
+}
+
+func (i *IdentifyNotifier) Send(update keybase1.CanonicalTLFNameAndIDWithBreaks) {
+	i.RLock()
+	tlfName := update.CanonicalName.String()
+	if stored, ok := i.identCache[tlfName]; ok {
+		// We have the exact update stored, don't send it again
+		if stored.Eq(update) {
+			defer i.RUnlock()
```

Comment:
I'm not a fan of two different `Unlocks` against one `Lock`.  Can you slightly refactor this code such as:

```go
i.RLock()
tlfName := update.CanonicalName.String()
stored, ok := i.identCache[tlfName]
i.RUnlock()
```